File: Projektowanie Obiektowe Oprogramowania/lista2/zad7.py
class Student:
    def __init__(self, name, courses):
        self.name = name
        self.courses = courses

    def getCourses(self):
        return self.courses


class Course:
    def __init__(self, name, grade):
        self.name = name
        self.grade = grade

    def getValue(self):
        return self.grade


class GradeController:
    def getStudentGradeInfo(self, student):
        grades = {}
        courses = student.getCourses()
        for course in courses:
            course_data = course.getAll(student)
            grade_value = course_data.getValue()
            grades[course.name] = grade_value
        return grades


class UsosWebPage:
    def __init__(self, grade_controller):
        self.grade_controller = grade_controller

    def show(self, student):
        student_grade_info = self.grade_controller.getStudentGradeInfo(student)
        print(f"Oceny studenta {student.name}:")
        for course_name, grade in student_grade_info.items():
            print(f" - {course_name}: {grade}")

File: Projektowanie Obiektowe Oprogramowania/lista2/zad7.cs
// dotnet script zad7.cs

var courses = new List<Course>
{
    new Course("Matematyka", 4.5),
    new Course("Fizyka", 3.8),
    new Course("Informatyka", 5.0)
};

var student = new Student("Jan Kowalski", courses);
var gradeController = new GradeController();
var usosWebPage = new UsosWebPage(gradeController);

usosWebPage.Show(student);

class Student
{
    public string Name { get; }
    public List<Course> Courses { get; }

    public Student(string name, List<Course> courses)
    {
        Name = name;
        Courses = courses;
    }

    public List<Course> GetCourses()
    {
        return Courses;
    }
}

class Course
{
    public string Name { get; }
    public double Grade { get; }

    public Course(string name, double grade)
    {
        Name = name;
        Grade = grade;
    }

    public double GetValue()
    {
        return Grade;
    }
}

class GradeController
{
    public Dictionary<string, double> GetStudentGradeInfo(Student student)
    {
        var grades = new Dictionary<string, double>();
        var courses = student.GetCourses();

        foreach (var course in courses)
        {
            grades[course.Name] = course.GetValue();
        }

        return grades;
    }
}

class UsosWebPage
{
    private readonly GradeController _gradeController;

    public UsosWebPage(GradeController gradeController)
    {
        _gradeController = gradeController;
    }

    public void Show(Student student)
    {
        var studentGradeInfo = _gradeController.GetStudentGradeInfo(student);
        Console.WriteLine($"Oceny studenta {student.Name}:");
        foreach (var entry in studentGradeInfo)
        {
            Console.WriteLine($" - {entry.Key}: {entry.Value}");
        }
    }
}
